### PR TITLE
Feature/hocs 832 add remove users to teams

### PR DESCRIPTION
**Pull request by leighkirky-zz**

### CODE REVIEW

File: src/shared/pages/team/teamView/teamView.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import { User } from '../../../models/user';
+import { GENERAL_ERROR_TITLE, LOAD_TEAM_ERROR_DESCRIPTION, LOAD_TEAM_MEMBERS_ERROR_DESCRIPTION } from '../../../models/constants';
+import ErrorSummary from '../../../common/components/errorSummary';
+
+interface MatchParams {
+    teamId: string;
+}
+
+interface TeamMembersProps extends RouteComponentProps<MatchParams> {
+    history: History;
```
Comment (lewismuirhead) [inline src/shared/pages/team/teamView/teamView.tsx]: I think the RouteComponentProps interface already includes 'history'

File: src/shared/pages/team/teamView/teamView.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                getTeamMembers(teamId)
+                    .then((users: User[]) => dispatch({ type: 'PopulateTeamMembers', payload: users }))
+                    .catch(() => {
+                        dispatch({ type: 'SetGeneralError', payload: { description: LOAD_TEAM_MEMBERS_ERROR_DESCRIPTION, title: GENERAL_ERROR_TITLE } });
+                    });
+            });
+    };
+
+    const DisplayTeamTable = () => (
+        <div>
```
Comment (lewismuirhead) [inline src/shared/pages/team/teamView/teamView.tsx]: are these two divs needed? or do they need to have the row/column classes?

File: src/shared/router/routes/index.ts [hunk 3]
```diff
@@ -15,6 +18,24 @@ const routes = [
         component: Dashboard,
         title: 'Dashboard'
     },
+    {
+        path: '/team_search',
```
Comment (lewismuirhead) [inline src/shared/router/routes/index.ts]: should the paths be kebab case?

File: src/shared/services/usersService.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    }
+}
+
+export const getUsers = () => new Promise((resolve, reject) => axios
+    .get('/api/users')
+    .then(response => resolve(response.data))
+    .catch(reason => reject(reason))
+);
+
+export const getTeamMembers = (teamId: string) => new Promise((resolve, reject) => axios
```
Comment (lewismuirhead) [inline src/shared/services/usersService.ts]: I think this should be in the teams service

**Review by lewismuirhead — CHANGES_REQUESTED** [review]

File: src/shared/pages/team/teamSearch/actions.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import ErrorMessage from '../../../models/errorMessage';
+
+export type SetTeams = {
+    type: 'SetTeams';
+    payload?: Item[];
+};
+
+export type AddTeamUUID = {
+    type: 'AddTeamUUID'
+    payload: any;
```
[the remark this answers is not in the document]
Comment (leighkirky-zz) [reply src/shared/pages/team/teamSearch/actions.tsx]: Removed

File: src/shared/pages/team/teamSearch/reducer.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { State } from './state';
+import { Action } from './actions';
+
+export const reducer = (state: State, action: Action): State => {
+    switch (action.type) {
+        case 'SetTeams':
+            // @ts-ignore
```
[the remark this answers is not in the document]
Comment (leighkirky-zz) [reply src/shared/pages/team/teamSearch/reducer.tsx]: removed

File: src/shared/pages/team/teamView/reducer.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { State } from './state';
+import { Action } from './actions';
+
+export const reducer = (state: State, action: Action): State => {
+    switch (action.type) {
+        case 'SetTeamName':
+            // @ts-ignore
```
[the remark this answers is not in the document]
Comment (leighkirky-zz) [reply src/shared/pages/team/teamView/reducer.tsx]: removed

File: src/shared/pages/team/teamView/teamView.tsx (hunk 1 above)
Comment (leighkirky-zz) [reply src/shared/pages/team/teamView/teamView.tsx]: Yes it does. removed

File: src/shared/pages/team/teamView/teamView.tsx (hunk 2 above)
Comment (leighkirky-zz) [reply src/shared/pages/team/teamView/teamView.tsx]: Divs removed ad wrapped in <Fragment>

File: src/shared/router/routes/index.ts (hunk 3 above)
Comment (leighkirky-zz) [reply src/shared/router/routes/index.ts]: Changed all paths in UI to kebab case

File: src/shared/services/usersService.ts (hunk 4 above)
Comment (leighkirky-zz) [reply src/shared/services/usersService.ts]: Moved to the teams service

**Comment by leighkirky-zz** [comment]
All resolved. @lewismuirhead do you want to eye over the changes before I merge?

**Review by lewismuirhead — APPROVED** [review]
looks good

**Pull request merged**